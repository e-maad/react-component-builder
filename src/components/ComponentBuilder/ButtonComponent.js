import React from 'react';

const ButtonComponent = ({ text, id }) => {
    return <button id={id} className='button-component'>{text}</button>
}

export default ButtonComponent;