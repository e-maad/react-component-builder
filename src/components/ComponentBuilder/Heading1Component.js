import React from 'react';

const Heading1Component = ({ text, id }) => {
    return <h1 id={id} className='heading1-component'>{text}</h1>
}

export default Heading1Component;