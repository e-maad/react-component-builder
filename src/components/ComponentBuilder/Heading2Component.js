import React from 'react';

const Heading2Component = ({ text, id }) => {
    return <h2 id={id} className='heading2-component'>{text}</h2>
}

export default Heading2Component;