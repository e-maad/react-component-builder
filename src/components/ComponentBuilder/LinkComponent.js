import React from 'react';

const LinkComponent = ({ text, url, id }) => {
    return <a id={id} href={url} target="_blank" className='link-component'>{text}</a>
}

export default LinkComponent;