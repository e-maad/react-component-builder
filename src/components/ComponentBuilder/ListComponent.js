import React from 'react';

const ListComponent = ({ li, id }) => {
    return <ul id={id} className='list-component'>
        {li ? li.map(item => <li key={item}>{item}</li>) : ''}
    </ul>
}

export default ListComponent;