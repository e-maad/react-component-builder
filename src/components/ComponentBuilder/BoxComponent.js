import React from 'react';
import ChildrenComponent from './ChildrenComponent';

const BoxComponent = ({ borderSize, children, id }) => {
    return <div id={id} className='box-component' style={{ borderWidth: borderSize }}>
        {children ? <ChildrenComponent children={children} /> : ''}
    </div>
}

export default BoxComponent;