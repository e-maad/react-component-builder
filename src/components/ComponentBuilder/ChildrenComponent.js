import React from 'react';
import ComponentBuilder from './ComponentBuilder';

const ChildrenComponent = ({ children }) => {
    return children ? Object.keys(children).map(key =>
        <ComponentBuilder component={children[key]} id={key} />
    ) : ''
}

export default ChildrenComponent;