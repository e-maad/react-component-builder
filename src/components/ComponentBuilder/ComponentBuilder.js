import React from 'react';
import BoxComponent from './BoxComponent';
import ButtonComponent from './ButtonComponent';
import Heading1Component from './Heading1Component';
import Heading2Component from './Heading2Component';
import LinkComponent from './LinkComponent';
import ListComponent from './ListComponent';
import ModalComponent from './ModalComponent';
import ParagraphComponent from './ParagraphComponent';

const ComponentBuilder = ({ component, id }) => {

    const getComponent = () => {
        switch (component.Content.type) {
            case 'H1Component':
                return <Heading1Component {...component.Content.props} id={id} />
            case 'H2Component':
                return <Heading2Component {...component.Content.props} id={id} />
            case 'ParagraphComponent':
                return <ParagraphComponent {...component.Content.props} id={id} />
            case 'ListComponent':
                return <ListComponent {...component.Content.props} id={id} />
            case 'ButtonComponent':
                return <ButtonComponent {...component.Content.props} id={id} />
            case 'LinkComponent':
                return <LinkComponent {...component.Content.props} id={id} />
            case 'BoxComponent':
                return <BoxComponent {...{ ...component.Content.props, children: component.Children }} id={id} />
            case 'ModalComponent':
                return <ModalComponent {...{ ...component.Content.props, children: component.Children }} id={id} />
            default:
                return;
        }
    }
    return getComponent()
}

export default ComponentBuilder;