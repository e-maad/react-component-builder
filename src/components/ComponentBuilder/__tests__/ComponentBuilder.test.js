import React from 'react';
import ChildrenComponent from '../ChildrenComponent';
import ComponentBuilder from '../ComponentBuilder';

describe('Component Builder Component', () => {

    it('match the snapshot', () => {
        const wrapper = renderer.create(<ChildrenComponent />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('must verify the H1 component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "H1Component",
                props: {
                    "text": "Our benefits"
                }
            }

        }} />).html().includes('heading1-component')).toBe(true);
    });

    it('must verify the H2 component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "H2Component",
                props: {
                    "text": "Our benefits"
                }
            }

        }} />).html().includes('heading2-component')).toBe(true);
    });

    it('must verify the paragraph component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "ParagraphComponent",
                props: {
                    "text": "Our benefits"
                }
            }

        }} />).html().includes('paragraph-component')).toBe(true);
    });

    it('must verify the list component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "ListComponent",
                props: {
                    "li": [
                        "free to use",
                        "superfast",
                        "and pretty, too!"
                    ]
                }
            }

        }} />).html().includes('list-component')).toBe(true);
    });

    it('must verify the button component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "ButtonComponent",
                props: {
                    "text": "Click me"
                }
            }

        }} />).html().includes('button-component')).toBe(true);
    });

    it('must verify the link component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "LinkComponent",
                props: {
                    "text": "Click me"
                }
            }

        }} />).html().includes('link-component')).toBe(true);
    });

    it('must verify the box component', () => {
        expect(shallow(<ComponentBuilder component={{
            Content: {
                type: "BoxComponent",
                props: {
                    "borderSize": "7px"
                }
            }

        }} />).html().includes('box-component')).toBe(true);
    });

});