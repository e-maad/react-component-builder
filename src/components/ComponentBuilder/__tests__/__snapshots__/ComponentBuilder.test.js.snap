// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Builder Component match the snapshot 1`] = `""`;
